### PR TITLE
Add client validation for organization CRUD operations

### PR DESCRIPTION
**Pull request by isxoqovxasanboy**

### CODE REVIEW

**Review by Ilxom-22 — DISMISSED** [review]

**Review by sulton-max — CHANGES_REQUESTED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs [hunk 1]
```diff
@@ -87,5 +96,22 @@ public IQueryable<Organization> Get(OrganizationFilter organizationFilter, Query
...
+            return ValueTask.FromResult<Organization?>(null);
+        }
+
+        return organizationRepository.DeleteByIdAsync(organizationId, commandOptions, cancellationToken);
+    }
+
+    private bool ValidateClient(Guid organizationId, Guid clientId)
+    {
+        var isFoundClient = organizationRepository.Get().Where(organization =>
+            organization.ClientId == clientId && organization.Id == organizationId);
```
Comment (Ilxom-22) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs:113]: You can put the predicate in the Where() method to the Get() method.

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/QueryHandlers/OrganizationGetQueryHandler.cs [hunk 2]
```diff
@@ -15,13 +15,11 @@ namespace Feedback.Analyzer.Infrastructure.Organizations.QueryHandlers;
...
 public class OrganizationGetQueryHandler(
     IMapper mapper,
-    IOrganizationService organizationService,
-    IRequestContextProvider requestContextProvider) : IQueryHandler<OrganizationGetQuery, ICollection<OrganizationDto>>
+    IOrganizationService organizationService) : IQueryHandler<OrganizationGetQuery, ICollection<OrganizationDto>>
 {
     public async Task<ICollection<OrganizationDto>> Handle(OrganizationGetQuery organizationGetQuery,
                                                            CancellationToken cancellationToken)
     {
-        organizationGetQuery.Filter.ClientId = requestContextProvider.GetUserId();
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/QueryHandlers/OrganizationGetQueryHandler.cs:24]: No, return this back

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs [hunk 3]
```diff
@@ -29,29 +31,35 @@ public class OrganizationService(
 
     public IQueryable<Organization> Get(OrganizationFilter organizationFilter, QueryOptions queryOptions = default)
     {
+        organizationFilter.ClientId = requestContextProvider.GetUserId();
+
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs:35]: Remove this

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs [hunk 4]
```diff
@@ -29,29 +31,35 @@ public class OrganizationService(
...
+            ? organizationRepository.GetByIdAsync(organizationId, queryOptions, cancellationToken)
+            : ValueTask.FromResult<Organization?>(null);
 
     public ValueTask<Organization> CreateAsync(
         Organization organization,
         CommandOptions commandOptions = default,
         CancellationToken cancellationToken = default)
     {
+        organization.ClientId = requestContextProvider.GetUserId();
+
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs:62]: Move this to commad handler

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs [hunk 5]
```diff
@@ -87,5 +96,22 @@ public IQueryable<Organization> Get(OrganizationFilter organizationFilter, Query
...
+        return organizationRepository.DeleteByIdAsync(organizationId, commandOptions, cancellationToken);
+    }
+
+    private bool ValidateClient(Guid organizationId, Guid clientId)
+    {
+        var isFoundClient = organizationRepository.Get(organization =>
+            organization.ClientId == clientId && organization.Id == organizationId);
+
+        return isFoundClient.Any();
+    }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs:116]: Remove private method for validation

**Review by sulton-max — CHANGES_REQUESTED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs [hunk 6]
```diff
@@ -29,25 +30,26 @@ public class OrganizationService(
...
         return organizationQuery;
     }
 
     public ValueTask<Organization?> GetByIdAsync(
         Guid organizationId,
         QueryOptions queryOptions = default,
         CancellationToken cancellationToken = default
-    ) 
+    )
         => organizationRepository.GetByIdAsync(organizationId, queryOptions, cancellationToken);
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs:50]: Add client filter

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs [hunk 7]
```diff
@@ -83,9 +88,16 @@ public IQueryable<Organization> Get(OrganizationFilter organizationFilter, Query
...
         CommandOptions commandOptions = default,
         CancellationToken cancellationToken = default)
-        => organizationRepository.DeleteByIdAsync(organizationId, commandOptions, cancellationToken);
+    {
+        var foundOrganization = await organizationRepository.GetByIdAsync(organizationId, cancellationToken: cancellationToken);  
+
+        if (foundOrganization!.ClientId != requestContextProvider.GetUserId())
+            throw new UnauthorizedAccessException();
+        
+        return await organizationRepository.DeleteByIdAsync(foundOrganization.Id, commandOptions, cancellationToken);
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs:101]: Add validation

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by sulton-max — APPROVED** [review]

**Pull request merged**